### PR TITLE
CORS-3220: Initiate CAPG installation

### PR DESCRIPTION
**Pull request by barbacbd**
** Added the base functions for the CAPG Installation 
** Allow the provider to be selected in ProviderForPlatform, when the feature gate exists 
** Updated the cluster to create two assets that would cause issues during install when they can not be retrieved

### CODE REVIEW

File: pkg/infrastructure/platform/platform.go [hunk 1]
```diff
@@ -52,6 +54,9 @@ func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastr
 	case baremetaltypes.Name:
 		return terraform.InitializeProvider(baremetal.PlatformStages), nil
 	case gcptypes.Name:
+		if fg.Enabled(configv1.FeatureGateClusterAPIInstall) {
+			return clusterapi.InitializeProvider(gcpcapi.Provider{}), nil
+		}
```
Comment (patrickdillon) [inline pkg/infrastructure/platform/platform.go:59]: can you add this exact same code to the platform_altinfra.go

File: pkg/infrastructure/gcp/clusterapi/clusterapi.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+var _ clusterapi.PreProvider = (*Provider)(nil)
+var _ clusterapi.IgnitionProvider = (*Provider)(nil)
+var _ clusterapi.InfraReadyProvider = (*Provider)(nil)
+var _ clusterapi.PostProvider = (*Provider)(nil)
+
+func (p Provider) Name() string {
+	return gcptypes.Name
+}
+
+func (p Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```
Comment (patrickdillon) [inline pkg/infrastructure/gcp/clusterapi/clusterapi.go:22]: nit: as we're not saving any state in the Provider, you might consider switching to use the unnamed provider:

```suggestion
func (_ Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```
I'm fine with saving state in the provider,  but we're not doing it yet.

**Comment by patrickdillon** [comment]
/test altinfra-gcp-capi-ovn
for funzies

File: pkg/infrastructure/gcp/clusterapi/clusterapi.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+// Name returns the name for the platform.
+func (_ Provider) Name() string { // nolint:stylecheck
+	return gcptypes.Name
+}
+
+// PreProvision is called before provisioning using CAPI controllers has initiated.
+// GCP resources that are not created by CAPG (and are required for other stages of the install) are
+// created here using the gcp sdk.
+func (_ Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error { // nolint:stylecheck,revive
```
Comment (patrickdillon) [inline pkg/infrastructure/gcp/clusterapi/clusterapi.go:26]:
```suggestion
func (Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```
Sorry didnt realize the linter would complain. Lets not no lint. It seems like the error message is suggesting to fix by omitting the variable name

File: pkg/infrastructure/platform/platform_altinfra.go [hunk 4]
```diff
@@ -22,6 +23,11 @@ func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastr
 	case azuretypes.Name:
 		panic("not implemented")
 		return nil, nil
+	case gcptypes.Name:
+		if fg.Enabled(configv1.FeatureGateClusterAPIInstall) {
```
Comment (bfournie) [inline pkg/infrastructure/platform/platform_altinfra.go:27]: It looks like a few imports are missing which is causing the altinfra-images test to fail
pkg/infrastructure/platform/platform_altinfra.go:27:17: undefined: configv1
pkg/infrastructure/platform/platform_altinfra.go:28:11: undefined: clusterapi
pkg/infrastructure/platform/platform_altinfra.go:28:41: undefined: gcpcapi

**Pull request merged**